homework1/main.py

```diff
@@ -0,0 +1,92 @@
+import pathlib
+import os
+import random
+import time
+import timeit
+import functools
+import requests
+
+import os, psutil
+
+# pid = os.getpid()
+
+
+
+
+
+class LFUCache:
+    def __init__(self, func, max_limit = 10):
+        self.func = func
+        self.cache = dict()
+        self.counter = dict()
+        self.max_limit = max_limit
+
+    def add(self, cache_key, result):
+        self.cache.update({cache_key:result})
+
+    def update_counter(self, cache_key):
+        if cache_key in self.cache:
+            self.counter[cache_key]+=1
+        else:
+            self.counter.update({cache_key:1})
```

Comment:
@LFUCache(max_limit=3)
---->>>

Traceback (most recent call last):
  File "/Users/mlazoryk/Library/Application Support/JetBrains/PyCharm2023.1/scratches/scratch.py", line 64, in <module>
    @LFUCache(max_limit=3)
TypeError: __init__() missing 1 required positional argument: 'func'